### PR TITLE
Automation code PO-652

### PR DESCRIPTION
**Pull request by rajanigandra**
### Jira link

See [PROJ-XXXXXX](https://tools.hmcts.net/jira/browse/PROJ-XXXXXX)

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: cypress/support/step_definitions/genericSteps/genericSteps.ts [hunk 1]
```diff
@@ -79,6 +79,11 @@ Then('I see {string} is {string}', (accountList: string, value: string) => {
 When('I reload the page', () => {
   cy.reload();
 });
+
+Then('I see {string} link', (buttonName: string) => {
+  cy.contains('a', buttonName).should('exist', buttonName);
+
```
Comment (CadeFaulkner) [inline cypress/support/step_definitions/genericSteps/genericSteps.ts:85]: can this be moved to the genericSteps/linkSteps.ts please?
Also the regex should be 'I see **the** {string} link

File: cypress/support/step_definitions/accountDetails/businessUnitAndDefendantType.ts [hunk 2]
```diff
@@ -56,8 +56,8 @@ When('I select parent or guardian to pay', () => {
 When('I select company', () => {
   cy.get('input[id="company"]').check().should('be.checked');
 });
-Then('I click on cancel', () => {
-  cy.contains('Cancel').click();
+Then('I click on {string}', (cancelLink: string) => {
+  cy.contains('a', cancelLink).click();
```
Comment (CadeFaulkner) [inline cypress/support/step_definitions/accountDetails/businessUnitAndDefendantType.ts:60]: there is already a step for this in genericSteps/linkSteps.ts

File: cypress/e2e/functional/opal/manualAccountCreation/welshLanguages/PO-465_languagePreferencesOnAccountDetails.feature [hunk 3]
```diff
@@ -23,6 +23,9 @@ Feature: PO-465 language preferences page for all defendant types
 
     Then I see "Document language" has a change link in the account details table
     And I see "Hearing language" has a change link in the account details table
+
+
+
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/welshLanguages/PO-465_languagePreferencesOnAccountDetails.feature:28]: Is there supposed to be a check in here?

File: cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature [hunk 4]
```diff
@@ -0,0 +1,150 @@
+Feature: PO-652 Move 'Cancel' button to beneath the navigation buttons
+
+#Covered parent or guardian scenario on PO-569, AC-6
+#Covered company scenarios on PO-365, AC6
+    Background:
+
+        Given I am on the OPAL Frontend
+        Then I see "Opal" in the header
+
+    Scenario: AC1 - Cancel button on Create account page, Personal details page, contact details page, Employer details page and account notes and comments
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature:10]: purely out of curiosity what was the decision for not integrating these checks into the other feature files

File: cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        Then I see the "Account comments and notes" section heading
+        And I see the "Account comments and notes" link under the "Account comments and notes" section
+        And I click on the "Account comments and notes" link
+        Then I see "Account comments and notes" on the page header
+
+        When I see "Cancel" link
+        Then I click on "Cancel"
+        Then I see "Account details" on the page header
+
+    Scenario: AC1 - Cancel button on Language preference for Create account page, Personal details page, contact details page, Employer details page and account notes and comments
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature:78]: purely out of curiosity what was the decision for not integrating these checks into the other feature files

File: cypress/support/step_definitions/accountDetails/businessUnitAndDefendantType.ts (hunk 2 above)
Comment (rajanigandra) [reply cypress/support/step_definitions/accountDetails/businessUnitAndDefendantType.ts:60]: There was already a step for cancel however it was not 'a' tag. In this ticket the selector was changed

File: cypress/support/step_definitions/genericSteps/genericSteps.ts (hunk 1 above)
Comment (rajanigandra) [reply cypress/support/step_definitions/genericSteps/genericSteps.ts:85]: okay

Comment (rajanigandra) [reply cypress/support/step_definitions/genericSteps/genericSteps.ts:85]: already in genericsteps.ts

File: cypress/support/step_definitions/accountDetails/businessUnitAndDefendantType.ts (hunk 2 above)
Comment (CadeFaulkner) [reply cypress/support/step_definitions/accountDetails/businessUnitAndDefendantType.ts:60]: In the genericSteps/linkSteps.ts file there are a couple existing link steps.
image

File: cypress/support/step_definitions/genericSteps/genericSteps.ts (hunk 1 above)
Comment (CadeFaulkner) [reply cypress/support/step_definitions/genericSteps/genericSteps.ts:85]: yep, so just needs to be moved from **_genericSteps.ts_** _**to genericSteps/linkSteps.ts**_

File: cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature (hunk 5 above)
Comment (rajanigandra) [reply cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature:78]: Showing Cancel button tests for few pages

File: cypress/support/step_definitions/accountDetails/businessUnitAndDefendantType.ts (hunk 2 above)
Comment (rajanigandra) [reply cypress/support/step_definitions/accountDetails/businessUnitAndDefendantType.ts:60]: Using already developed step now

File: cypress/e2e/functional/opal/manualAccountCreation/welshLanguages/PO-465_languagePreferencesOnAccountDetails.feature (hunk 3 above)
Comment (rajanigandra) [reply cypress/e2e/functional/opal/manualAccountCreation/welshLanguages/PO-465_languagePreferencesOnAccountDetails.feature:28]: deleted few steps

File: cypress/support/step_definitions/accountDetails/businessUnitAndDefendantType.ts (hunk 2 above)
Comment (CadeFaulkner) [reply cypress/support/step_definitions/accountDetails/businessUnitAndDefendantType.ts:60]: great, can this step please be removed as it is a duplicate of the existing step?

File: cypress/e2e/functional/opal/manualAccountCreation/accountCreationPage/PO-346_Business_Unit_And_Defendant_type.feature [hunk 6]
```diff
@@ -142,7 +142,7 @@ Feature: PO-346 business unit and defendant type Users associated to one and onl
     When I select the "Fine" radio button
     And I see the heading under the fine radio button is "Defendant type"
     And I see "If sole trader, choose 'Adult or youth only'" below the defendant type subheading
-    And I click on cancel
+    And I click on "Cancel"
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/accountCreationPage/PO-346_Business_Unit_And_Defendant_type.feature:145]: This needs updating to use the existing step

File: cypress/e2e/functional/opal/manualAccountCreation/accountCreationPage/PO-346_Business_Unit_And_Defendant_type.feature [hunk 7]
```diff
@@ -153,7 +153,7 @@ Feature: PO-346 business unit and defendant type Users associated to one and onl
     And I see "If sole trader, choose 'Adult or youth only'" below the defendant type subheading
 
     And I select the "Adult or youth only" radio button
-    And I click on cancel
+    And I click on "Cancel"
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/accountCreationPage/PO-346_Business_Unit_And_Defendant_type.feature:156]: This needs updating to use the existing step

File: cypress/support/step_definitions/accountDetails/businessUnitAndDefendantType.ts (hunk 2 above)
Comment (CadeFaulkner) [reply cypress/support/step_definitions/accountDetails/businessUnitAndDefendantType.ts:60]: This step can be removed as the existing step is being used.

File: cypress/e2e/functional/opal/manualAccountCreation/companyDetails/PO-365_companyDetails_company.feature [hunk 8]
```diff
@@ -134,6 +136,11 @@ Feature: PO-365 Creating a fines account for a Company where defendant type is c
     And I select "Remove" link below the "Alias 2", "Company name" input
     And I see "Add another alias" button
 
+    #verifying cancel button which is developed on PO-652
+    When I see the "Cancel" link
+    Then I click on "Cancel"
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/companyDetails/PO-365_companyDetails_company.feature:141]: This needs updating to use the existing step

File: cypress/e2e/functional/opal/manualAccountCreation/adultOrYouthWithParentOrGuardian/PO-569_parentOrGuardianScreen.feature [hunk 9]
```diff
@@ -106,6 +110,11 @@ Feature: PO-569 PO-678 Updates to Parent or guardian details screen
         Then I no longer see "Alias 2" sub heading
         Then I see the "Alias 1" sub heading in aliases
 
+        #verifying cancel button which is developed on PO-652
+        When I see the "Cancel" link
+        Then I click on "Cancel"
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/adultOrYouthWithParentOrGuardian/PO-569_parentOrGuardianScreen.feature:115]: This needs updating to use the existing step

File: cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+        Then I click on "Cancel"
+        Then I see "Account details" on the page header
+
+        Then I see the "Account comments and notes" section heading
+        And I see the "Account comments and notes" link under the "Account comments and notes" section
+        And I click on the "Account comments and notes" link
+        Then I see "Account comments and notes" on the page header
+
+        When I see the "Cancel" link
+        Then I click on "Cancel"
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature:140]: This needs updating to use the existing step

File: cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+        Then I see "Account details" on the page header
+
+        Then I see the "Defendant details" section heading
+        And I see the "Employer details" link under the "Defendant details" section
+        And I click on the "Employer details" link
+        Then I see "Employer details" on the page header
+
+        When I see the "Cancel" link
+        Then I click on "Cancel"
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature:131]: This needs updating to use the existing step

File: cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+        Then I see "Account details" on the page header
+
+        Then I see the "Defendant details" section heading
+        And I see the "Contact details" link under the "Defendant details" section
+        And I click on the "Contact details" link
+        Then I see "Defendant contact details" on the page header
+
+        When I see the "Cancel" link
+        Then I click on "Cancel"
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature:122]: This needs updating to use the existing step

File: cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+        Then I see "Account details" on the page header
+
+        Then I see the "Defendant details" section heading
+        And I see the "Personal details" link under the "Defendant details" section
+        And I click on the "Personal details" link
+        Then I see "Personal details" on the page header
+
+        When I see the "Cancel" link
+        Then I click on "Cancel"
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature:113]: This needs updating to use the existing step

File: cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+        And I see the "Document language" is "Welsh and English" in the account details table
+        And I see the "Hearing language" is "Welsh and English" in the account details table
+
+        Then I see the "Court details" section heading
+        And I see the "Court details" link under the "Court details" section
+        Then I see the status of "Court details" is "Not provided"
+        And I click on the "Court details" link
+
+        When I see the "Cancel" link
+        Then I click on "Cancel"
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature:104]: This needs updating to use the existing step

File: cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature [hunk 15]
```diff
@@ -0,0 +1,150 @@
...
+        Then I see "Account details" on the page header
+
+    Scenario: AC1 - Cancel button on Language preference for Create account page, Personal details page, contact details page, Employer details page and account notes and comments
+        When I sign in as "opal-test-6@HMCTS.NET"
+        Then I am on the dashboard
+        When I navigate to Manual Account Creation
+
+        Then I see "Business unit and defendant type" on the page header
+        When I see the "Cancel" link
+        Then I click on "Cancel"
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature:85]: This needs updating to use the existing step

File: cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+        Then I see "Account details" on the page header
+
+        Then I see the "Account comments and notes" section heading
+        And I see the "Account comments and notes" link under the "Account comments and notes" section
+        And I click on the "Account comments and notes" link
+        Then I see "Account comments and notes" on the page header
+
+        When I see the "Cancel" link
+        Then I click on "Cancel"
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature:75]: This needs updating to use the existing step

File: cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature [hunk 17]
```diff
@@ -0,0 +1,150 @@
...
+        Then I see "Account details" on the page header
+
+        Then I see the "Defendant details" section heading
+        And I see the "Employer details" link under the "Defendant details" section
+        And I click on the "Employer details" link
+        Then I see "Employer details" on the page header
+
+        When I see the "Cancel" link
+        Then I click on "Cancel"
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature:66]: This needs updating to use the existing step

File: cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature [hunk 18]
```diff
@@ -0,0 +1,150 @@
...
+        Then I see "Account details" on the page header
+
+        Then I see the "Defendant details" section heading
+        And I see the "Contact details" link under the "Defendant details" section
+        And I click on the "Contact details" link
+        Then I see "Defendant contact details" on the page header
+
+        When I see the "Cancel" link
+        Then I click on "Cancel"
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature:57]: This needs updating to use the existing step

File: cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature [hunk 19]
```diff
@@ -0,0 +1,150 @@
...
+        Then I see "Account details" on the page header
+
+        Then I see the "Defendant details" section heading
+        And I see the "Personal details" link under the "Defendant details" section
+        And I click on the "Personal details" link
+        Then I see "Personal details" on the page header
+
+        When I see the "Cancel" link
+        Then I click on "Cancel"
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature:48]: This needs updating to use the existing step

File: cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature [hunk 20]
```diff
@@ -0,0 +1,150 @@
...
+        And I click the "Continue" button
+        Then I see "Account details" on the page header
+
+        Then I see the "Court details" section heading
+        And I see the "Court details" link under the "Court details" section
+        Then I see the status of "Court details" is "Not provided"
+        And I click on the "Court details" link
+
+        When I see the "Cancel" link
+        Then I click on "Cancel"
```
Comment (CadeFaulkner) [inline cypress/e2e/functional/opal/manualAccountCreation/accountDetails/PO-652_CancelButton.feature:39]: This needs updating to use the existing step

File: cypress/support/step_definitions/genericSteps/genericSteps.ts (hunk 1 above)
Comment (rajanigandra) [reply cypress/support/step_definitions/genericSteps/genericSteps.ts]: deleted this step in the last commit

File: cypress/support/step_definitions/accountDetails/businessUnitAndDefendantType.ts (hunk 2 above)
Comment (rajanigandra) [reply cypress/support/step_definitions/accountDetails/businessUnitAndDefendantType.ts:60]: the step using, deleted that I've created

File: cypress/e2e/functional/opal/manualAccountCreation/accountCreationPage/PO-346_Business_Unit_And_Defendant_type.feature (hunk 6 above)
Comment (rajanigandra) [reply cypress/e2e/functional/opal/manualAccountCreation/accountCreationPage/PO-346_Business_Unit_And_Defendant_type.feature:145]: updated and pushed code

**Review by CadeFaulkner — APPROVED** [review]

**Review by Alfred-Gillingham2 — APPROVED** [review]

**Pull request merged**